src/jarabe/main.py

```diff
@@ -347,6 +347,50 @@ def setup_fonts():
     settings.set_property("gtk-font-name", "%s %f" % (face, size))
 
 
+def setup_proxy():
+    g_mode = Gio.Settings('org.sugarlabs.system.proxy').get_string('mode')
+    if g_mode == 'manual':
+        g_http = Gio.Settings('org.sugarlabs.system.proxy.http')
+        g_http_host = g_http.get_string('host')
+        g_http_port = str(g_http.get_int('port'))
+        if(g_http.get_boolean('use-authentication')):
+            text_to_set = "http://%s:%s@%s:%s/" % (
```

Comment:
Wouldn't it make sense to also have the authentication with https proxies?

Disclaimer:  never used a proxy in my life.